source/MaterialXRenderGlsl/TextureBaker.cpp

```diff
@@ -215,7 +215,7 @@ void TextureBaker::writeBakedMaterial(const FilePath& filename, const StringVec&
 
             // Add the image node.
             NodePtr bakedImage = bakedNodeGraph->addNode("image", bindInput->getName() + "_baked", bindInput->getType());
-            ParameterPtr param = bakedImage->addParameter("file", "filename");
+            InputPtr param = bakedImage->addInput("file", "filename");
```

Comment:
minor: should this no longer be called param?

Reply:
Should it be set as uniform as well as per my previous comment?

Reply:
Maybe we should add a `uniform` boolean to the `addInput` method, so we don't have to call two methods to create the input. And this flag could be false by default.